src/asm/fstack.cpp

```diff
@@ -605,7 +619,7 @@ void fstk_Init(char const *mainPath, size_t maxDepth)
 
 	// Check that max recursion depth won't allow overflowing node `malloc`s
 	// This assumes that the rept node is larger
-#define DEPTH_LIMIT ((SIZE_MAX - sizeof(struct FileStackReptNode)) / sizeof(uint32_t))
+#define DEPTH_LIMIT (SIZE_MAX / sizeof(uint32_t))
```

Comment:
@ISSOtm Please confirm if I updated this check correctly.

Reply:
Do we still `malloc` the nodes manually? If we're using STL containers (probably a `vector`?), I would assume that they perform the check themselves.

Reply:
Okay, I see -- yes, we're using a `std::vector` now instead of a FAM, so there's no danger of `malloc`ing an overflowed manually computed size.